ARCs/arc-0060.md

```diff
@@ -152,298 +182,206 @@ export interface HDWalletMetadata {
+    encodedData: string
+} => T;
 ```
 
-- The `ARC60Domain` object indicates the domain separator to be used for signing. It **SHOULD** be set to `"arc60"` or empty `""`. Algorand transaction domain separators `TX` and `TG` **MUST** be forbidden.
-- The `bytes` object is the byte array of data being signed. In the example, it is a 32-byte challenge to be signed for authentication scope. The `bytes` field cannot be a valid Algorand transaction, therefore it **MUST NOT** be prepended with a known domain separator `TX` or `TG`.
+#### Function `validate`
 
-> Algorand domain separators can be found in the <a href="https://github.com/algorandfoundation/specs/blob/master/dev/crypto.md#domain-separation">Algorand specs</a> and in the <a href="https://github.com/algorand/go-algorand/blob/master/protocol/hash.go#L21">Go type HashID</a>.
+Function `validate` takes an instance of the structured data `T` as input and checks if it is valid according to a set of rules that are individual to each scope. For example, a validation function can check if the data has a certain size, if a domain matches, etc. Each scope must come with a precise set of validation rules that allows wallets to determine if the input is valid or not.
 
-A valid `StdDataStr` object for `ScopeType=AUTH` should look like the following:
+```ts
+export type ValidateFunction = {
+    structuredData: T
+} => boolean;
+```
 
-```json
-{
-  "ARC60Domain" : "arc60",
-  "bytes" : "[99, 50, 97, 57, 51, 99, ..., 57, 49, 53]"
-}
+#### Function `serialize`
+
+Function `serialize` flattens an instance of the structured data `T` into a byte array that can be fed into a  signing function. It is important that this function is deterministic, meaning that two parties that serialize the same data receive the same byte array. This ensures that given an instance of `T`, a signer and a verifier operate on the same byte string.
+
+```ts
+export type SerializeFunction = {
+    structuredData: T
+} => Uint8Array;
 ```
 
-#### Error interface `SignDataError`
+#### Scope `MX_RANDOM`
 
-The `SignDataError` object extends the `SignTxnsError` defined in [ARC-1](./arc-0001.md).
+This scope is inspired by the `signBytes` function in the Algorand SDK which is currently used by some applications. The `signBytes` function allows arbitrary data signing with the only restriction that the data is neither a valid transaction nor a logic signature. This is achieved by enforcing that the first two bytes are the domain separator `MX`. Apart from this restriction, the data is an opaque, unstructured byte string.
+
+Functions `parse`, `validate`, and `serialize` can be defined as follows:
 
 ```ts
-export interface SignDataError extends Error {
-  code: number;
-  data?: any;
-  failingSignData: (StdSigData | null);
+function parse(data: string): Uint8Array {
+    return Buffer.from(data, 'base64');
+}
+function validate(data: Uint8Array): boolean {
+    return data.length >= 2 && data[0] === 0x4D && data[1] === 0x58;
```

Comment:
The length should be strictly greater than two IMO. I won't allow users to sign just the prefix `MX`.  WDYT?

Reply:
makes sense! I was also wondering if we should change the way `MX_RANDOM` works to mirror what the Algorand SDK does (https://github.com/algorand/js-algorand-sdk/blob/develop/src/main.ts#L68).

In this PR we have:
- `validate`: check if `data` starts with `MX`
- `serialize`: returns `data` as is

If the do it more like the Algorand SDK does it, we'd have to do:
- `validate`: no-op
- `serialize`: `concat('MX', data)`